### ISSUE TITLE
Scope for .NET naming conventions for EditorConfig

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#40764]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/idea/842310/scope-for-net-naming-conventions-for-editorconfig.html)._

---
<p>Is it possible to distinguish the scope for .editorconfig naming rules?</p> <p>I'd like to allow public fields in structs:</p><pre><code>public struct MyStruct {     public int StructField; } </code></pre> <p>But disallow public fields in classes:</p><pre><code>public class MyClass {     public int ClassField; } </code></pre> <p>Is this possible using .editorconfig naming rules?</p> <p>Back link: <a href="https://github.com/MicrosoftDocs/visualstudio-docs/issues/4400" target=_blank>https://github.com/MicrosoftDocs/visualstudio-docs/issues/4400</a></p>

---
#### Visual Studio Feedback System on 12/4/2019, 00:30 AM: 

Thank you for taking the time to provide your suggestion.  We will do some preliminary checks to make sure we can proceed further.  We'll provide an update once the issue has been triaged by the product team.
#### Kendra Havens [MSFT] on 1/3/2020, 06:49 PM: 

<p>Currently, you are not able to prefer private or public fields per type via the EditorConfig. In your second example for disallowing public fields in classes, would you want it to show a warning or just a message?</p>
<p>Thank you for helping Visual Studio improve!</p>

#### Luke on 1/3/2020, 11:48 PM: 

<p>I'd prefer a warning (which can be configured to be considered as an error which can prevent the project from being built).</p>

#### Kendra Havens on 1/6/2020, 11:14 AM: 

<p>Thank you for the additional info. I will move this ticket to our open source GitHub repo for further design discussion with the community.</p>

**Comment by CyrusNajmabadi** [comment]
No, it is not.  but it would be trivial for a customer to write an analyzer to do such a thing.  It would even make a very nice and simple sample for analyzers for people to look at.
